daemon/cmd/endpoint_restore.go

```diff
@@ -265,6 +341,11 @@ func (r *endpointRestorer) RestoreOldEndpoints() {
 
 	r.logger.Info("Restoring endpoints...")
 
+	// Validate that endpoints are compatible with the current datapath mode
+	if err := r.validateDatapathModeCompatibility(r.restoreState.possible); err != nil {
+		logging.Fatal(r.logger, "Datapath mode incompatibility detected", logfields.Error, err)
```

Comment:
Endpoint restore happens as part of daemon start hook. Should we return an error here and exit cleanly by failing hive start?

Reply:
👍 Updated to bubble up the error all the way to the hive start hook instead of panicing via `logging.Fatal`